lib/scoped/update.js

```diff
@@ -1,11 +1,57 @@
 module.exports = update
 
 var find = require('./find')
+var merge = require('lodash/merge')
 
 function update (type, api, objectsOrIds, change) {
   return find(type, api, objectsOrIds)
 
   .then(function (storedObjects) {
-    return api.update(objectsOrIds, change)
+    // objectsOrIds may contain updated information for the respective storedObject
+    // (e.g. _deleted = true), so incorporate all information in the objectOrId into
+    // the storedObject
+    var updatedObjects = updateObjects(objectsOrIds, storedObjects)
+
+    return api.update(updatedObjects, change)
+
+    .then(function (apiUpdates) {
+      if (!Array.isArray(apiUpdates)) {
+        return apiUpdates
+      }
+
+      return restoreOriginalFindErrors(updatedObjects, apiUpdates)
+    })
+  })
+}
+
+function updateObjects (objectsOrIds, storedObjects) {
+  if (!Array.isArray(objectsOrIds)) {
+    return updateObject(objectsOrIds, storedObjects)
+  }
+
+  return storedObjects.map(function (storedObject, index) {
+    if (storedObject instanceof Error) {
+      return storedObject
+    }
+
+    return updateObject(objectsOrIds[index], storedObject)
+  })
+}
+
+function updateObject (objectOrId, storedObject) {
+  if (typeof objectOrId !== 'object') {
+    return storedObject
+  }
+
+  return merge(storedObject, objectOrId)
+}
+
+function restoreOriginalFindErrors (updatedObjects, apiUpdates) {
```

Comment:
Hmm yeah that is a tricky one! 

I’d suggest we filter out the errors that `.find()` gives us and then re-insert them at the same place before `.update()` resolves. That way we don’t need to alter the implementation of the global `store.update()` method